pasteit/routes/api.js

```diff
@@ -0,0 +1,25 @@
+var express = require("express");
+var router = express();
+var uuid = require("uuid4");
+var fn = require("./functions")
+
+router.get("/api", function(req, res) {
+    res.json({ message: 'Hooray! Welcome to the RESTful-API' });
+});
+
+router.route("/api/create")
+    .post(function(req, res) {
+        var hash = fn.newPaste(req);
+        
+        res.json({ paste_url : req.get("host") + "/p/"+hash });    
+});
+
+router.route("/api/p/")
+    .post(function(req, res) {
+        var result = fn.getPaste(req.params.hash);
+
+
+        res.json({ error : "Invalid hash" }) ? result == null : res.json({ result : paste });
```

Comment:
No space between the key and the value in object definitions. No space between the closing and ending brackets in inline object declarations. Use triple equals for strict type comparison of null.

This whole expression is confusing. At first glance it seems to do nothing, until you read the documentation for res.json. https://expressjs.com/en/4x/api.html#res.json. Consider the following in terms of readability.

``` javascript
 // res.json({ error : "Invalid hash" }) ? result === null : res.json({ result : paste });

  if (result == null) {
    res.status(404).send("Not found"); // if the hash didn't exist then the URL is invalid, return a 404
  } else {
    res.json({"result": paste});
  }
```